Sources/DataStructures/Graph/Protocols/GraphProtocol.swift

```diff
@@ -62,6 +62,25 @@ extension GraphProtocol {
     public func neighbors(of source: Node, in nodes: Set<Node>? = nil) -> Set<Node> {
         return (nodes ?? self.nodes).filter { edges.contains(Edge(source,$0)) }
     }
+    
+    /// - Returns: A set of edges that contain the given `node` (either incident or
+    /// outgoing).
+    @inlinable
+    public func edges(containing node: Node) -> Set<Edge> {
```

Comment:
Could it be more performant to return `edges.contains { $0.contains(node) }` ?

IIUC, the current implementation runs two `O(*n*)` operations, then the `union`, whereas the above should just be a single `O(*n*)` pass.

Reply:
That said, `WeightedGraphProtocol`-conforming types, may have `O(*n*)` to get `edges` from `adjacents`. Hmm.

Reply:
`GraphProtocol.edges` _could_ be implemented as an `associatedype`.

```Swift
protocol GraphProtocol {
    associatedtype Edges: Sequence where Element == Edge
    var edges: Edges
}
```

Then, potentially `WeightedGraphProtocol` and `UnweightedGraphProtocol` could specialize this.

`WeightedGraphProtocol` could return a `LazySequence<...>`, while `UnweightedGraphProtocol` could return a `Set`.

This may not be worth all of this effort, though :).

Reply:
I think this is good for now, but food for thought for later!